src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -37,11 +38,25 @@ void HandleRestart(Layer * systemLayer, void * appState)
 {
     esp_restart();
 }
+
+void PostOTAStateChangeEvent(DeviceLayer::OtaState newState)
+{
+    DeviceLayer::ChipDeviceEvent otaChange;
+    otaChange.Type                     = DeviceLayer::DeviceEventType::kOtaStateChanged;
+    otaChange.OtaStateChanged.newState = newState;
+    CHIP_ERROR error                   = DeviceLayer::PlatformMgr().PostEvent(&otaChange);
+
```

Comment:
None of the OTA events require any kind of status/data to be sent?

Reply:
These are simple state changes event and may not require additional data.